File: nodejs/250124_S5_13301.js
/**
 * https://www.acmicpc.net/problem/13301
 */
const n = Number(require('fs').readFileSync(0, 'utf-8').toString().trim());
const dp = new Array(n+1).fill(0);
dp[1] = 1;
for (let i = 2; i <= n; i++) {
	dp[i] = dp[i-1] + dp[i-2];
}
console.log((dp[n] + dp[n] + dp[n-1]) * 2)

File: python/250124_S5_13301.py
# https://www.acmicpc.net/problem/13301

n = int(input())
dp = [0,1,1,2]
for i in range(4, n+1):
    dp.append(dp[i-1] + dp[i-2])
print((dp[n] * 4) + (dp[n-1] * 2))
